src/VisualStudio/Core/Def/Implementation/EditAndContinue/VisualStudioActiveStatementProvider.cs

```diff
@@ -122,6 +145,7 @@ void CancelWork()
                                             // the last active statement of the last runtime has been processed:
                                             if (Interlocked.Decrement(ref pendingRuntimes) == 0)
                                             {
+                                                completed = true;
                                                 completion.SetResult(builders.ToFlattenedImmutableArrayAndFree());
```

Comment:
You'll want this to be TrySetResult* probably, lest you have some race that is racing against TrySetException(). (As a paranoid concern, the CompletionRoutine were to run racing against this, it might run before completed is set to true.)